fetchai/ledger/contract.py

```diff
@@ -150,11 +172,8 @@ def _api(api: ContractsApiLike):
     def _from_json_object(obj):
         assert obj['version'] == 1
         source = base64.b64decode(obj['source']).decode()
-        sc = Contract(source)
-
         owner = obj['owner']
-        if owner is not None:
-            sc.owner = owner
+        sc = Contract(source, owner)
```

Comment:
Since this function is used in dumping and saving the contract to disk, with this change we will need to support constructing the contract with a predetermined nonce

Reply:
It is also worth adding a test for this case so that we can ensure that this happens in the future